examples_apps/run_state_learner.py

```diff
@@ -0,0 +1,453 @@
+
+-  :math:`\Phi(\lambda_i)` is a single mode non-Gaussian operation, in
+   this case chosen to be the Kerr interaction
+   :math:`\mathcal{K}(\kappa_i)=e^{i\kappa_i\hat{n}^2}` of strength
+   :math:`\kappa_i`.
+
+
+Hyperparameters
+---------------
+
+First, we must define the **hyperparameters** of our layer structure:
+
+-  ``cutoff``: the simulation Fock space truncation we will use in the
+   optimization. The TensorFlow backend will perform numerical
+   operations in this truncated Fock space when performing the
+   optimization.
+
+-  ``depth``: The number of layers in our variational quantum
+   circuit. As a general rule, increasing the number of layers (and
+   thus, the number of parameters we are optimizing over) increases the
+   optimizer's chance of finding a reasonable local minimum in the
+   optimization landscape.
+
+-  ``reps``: the number of steps in the optimization routine performing
+   gradient descent
+
+Some other optional hyperparameters include:
+
+-  The standard deviation of initial parameters. Note that we make a
+   distinction between the standard deviation of *passive* parameters
+   (those that preserve photon number when changed, such as phase
+   parameters), and *active* parameters (those that introduce or remove
+   energy from the system when changed).
+"""
+
+import numpy as np
+
+import strawberryfields as sf
+from strawberryfields.ops import *
+from strawberryfields.utils import operation
+
+# Cutoff dimension
+cutoff = 9
+
+# Number of layers
+depth = 15
+
+# Number of steps in optimization routine performing gradient descent
+reps = 200
+
+# Learning rate
+lr = 0.05
+
+# Standard deviation of initial parameters
+passive_sd = 0.1
+active_sd = 0.001
+
+######################################################################
+# The layer parameters :math:`\vec{\theta}`
+# -----------------------------------------
+#
+# We use TensorFlow to create the variables corresponding to the gate
+# parameters. Note that we focus on a single mode circuit where
+# each variable has shape ``(depth,)``, with each
+# individual element representing the gate parameter in layer :math:`i`.
+
+import tensorflow as tf
+
+# set the random seed
+tf.random.set_seed(42)
+
+# squeeze gate
+sq_r = tf.random.normal(shape=[depth], stddev=active_sd)
+sq_phi = tf.random.normal(shape=[depth], stddev=passive_sd)
+
+# displacement gate
+d_r = tf.random.normal(shape=[depth], stddev=active_sd)
+d_phi = tf.random.normal(shape=[depth], stddev=passive_sd)
+
+# rotation gates
+r1 = tf.random.normal(shape=[depth], stddev=passive_sd)
+r2 = tf.random.normal(shape=[depth], stddev=passive_sd)
+
+# kerr gate
+kappa = tf.random.normal(shape=[depth], stddev=active_sd)
+
+
+######################################################################
+# For convenience, we store the TensorFlow variables representing the
+# weights as a tensor, as well as creating the corresponding
+# single-mode Strawberry Fields program and parameters:
```

Comment:
This part of the comment refers to the next section now (the code parts were moved if I'm not mistaken)
```suggestion
# weights as a tensor:
```

Reply:
Good catch, I'll move it earlier!